### PR TITLE
Include option for including links to PRs with no release notes

### PR DESCRIPTION
**Pull request by Br4nd0R**
# Changes
* The `github_release_notes` task now has an `include_empty` option to for the ability to include links to PRs that have no release notes.

### CODE REVIEW

File: cumulusci/tasks/release_notes/generator.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
 class BaseReleaseNotesGenerator(object):
     def __init__(self):
         self.change_notes = []
+        self.empty_pull_requests = []
```
Comment (davisagli) [inline cumulusci/tasks/release_notes/generator.py]: Thinking about the naming here, we should probably call this `empty_change_notes` for symmetry with the existing `change_notes` list. Pull requests are the source of the change notes in the case of the github release notes generator, but this base class doesn't assume that's where they come from.

File: cumulusci/tasks/release_notes/generator.py [hunk 2]
```diff
@@ -136,6 +146,21 @@ def _get_release(self):
...
+            section_lines.append("\n# Pull requests with no release notes")
+            for content in self.empty_pull_requests:
+                section_lines.append(
+                    "\n* {}".format(self._mark_down_link_to_pr(content))
+                )
+
+        return section_lines
+
+    def _mark_down_link_to_pr(self, pull_request):
+        if pull_request:
```
Comment (davisagli) [inline cumulusci/tasks/release_notes/generator.py]: When would this line be false?

File: cumulusci/tasks/release_notes/generator.py [hunk 3]
```diff
@@ -136,6 +146,21 @@ def _get_release(self):
...
+            return "[[PR{}]({})]".format(pull_request.number, pull_request.html_url)
```
Comment (davisagli) [inline cumulusci/tasks/release_notes/generator.py]: Do you think we should include the PR title?

File: cumulusci/tasks/release_notes/generator.py [hunk 4]
```diff
@@ -186,8 +211,10 @@ def _update_release_content(self, release, content):
                 if parser_content and not parser.replaced:
                     new_body.append(parser_content + "\r\n")
 
-            content = u"\r\n".join(new_body)
-
+        # add empty PRs section
+        if self.include_empty_pull_requests:
+            new_body.extend(self._render_empty_pr_section())
```
Comment (davisagli) [inline cumulusci/tasks/release_notes/generator.py]: If the task is run twice we'll end up with 2 copies of this list in the release notes. That's probably not a huge problem since typically this task is only run once by automation. But worth being aware of that difference from how the other parser sections are handled.

File: cumulusci/tasks/release_notes/task.py [hunk 5]
```diff
@@ -25,6 +25,9 @@ class GithubReleaseNotes(BaseGithubTask):
             )
         },
         "publish": {"description": "Publish to GitHub release if True (default=False)"},
+        "include_empty": {
+            "description": "If True, include links to PRs that have no release notes (default=False)"
+        },
```
Comment (davisagli) [inline cumulusci/tasks/release_notes/task.py]: Don't forget to adjust the release_beta flow in cumulusci.yml to set this to True.

File: cumulusci/tasks/release_notes/tests/test_generator.py [hunk 6]
```diff
@@ -122,6 +122,18 @@ def test_init_with_last_tag(self):
...
+    def test_mark_down_link_to_pr(self):
+        pr = mock.Mock(number=1, html_url="http://pr", body="# Changes\r\n\r\nfoo")
+        github_info = self.github_info.copy()
+        self.mock_util.mock_get_repo()
+        generator = GithubReleaseNotesGenerator(
+            self.gh, github_info, PARSER_CONFIG, self.current_tag, self.last_tag
+        )
+        actual_link = generator._mark_down_link_to_pr(pr)
+        expected_link = "[[PR{}]({})]".format(pr.number, pr.html_url)
+        self.assertEquals(expected_link, actual_link)
```
Comment (davisagli) [inline cumulusci/tasks/release_notes/tests/test_generator.py]: We should probably also add a test that is more of an integration test (but still using faked github data) showing that the generator will create the empty PRs section if there are PRs with no other change notes.

Comment (davisagli) [reply cumulusci/tasks/release_notes/tests/test_generator.py]: You can see the missing test coverage here: https://coveralls.io/builds/25062620/source?filename=cumulusci%2Ftasks%2Frelease_notes%2Fgenerator.py#L150

File: cumulusci/tasks/release_notes/generator.py (hunk 2 above)
Comment (Br4nd0R) [reply cumulusci/tasks/release_notes/generator.py]: I don't think it ever would be. Removing :)

File: cumulusci/tasks/release_notes/generator.py (hunk 3 above)
Comment (Br4nd0R) [reply cumulusci/tasks/release_notes/generator.py]: I think it would help, especially since it would indicate to a QA/QE what exactly the scope of work was.

**Comment by Br4nd0R** [comment]
@davisagli - The added tests now provide full coverage for the new code; however, I'm not sure I've satisfied the "integration" test that you mentioned earlier. I was attempting to have a test where I can setup mocked PR data and utilize the `GithubReleaseNotesGenerator.__call__` method but we don't currently have the endpoint that `GithubChangeNotesProvider._get_pull_requests()` calls out to mocked under `cumulusci/tasks/release_notes/tests/utils.py --> MockUtil`. 

I can definitely make a mock for the endpoint but I wasn't quite sure how to tie-together pull requests generated by `MockUtil.mock_pull_request` to be returned by the newly mocked endpoint.

Would love to hear your thoughts, and definitely happy to implement this as it would be nice to test all of the parts as a whole.

**Comment by davisagli** [comment]
@Br4nd0R I think you can add the mock response for _get_pull_requests something like this:

        responses.add("GET", "{}/pulls".format(mock_util.repo_url), json=[pr1, pr2])

But I think your existing tests do a good job of testing the pieces and a full integration test isn't super important (especially since the piece that that would add of interacting with github is mostly mocked out anyway).
